src/modules/SX126x/SX126x.cpp

```diff
@@ -757,7 +775,7 @@ int16_t SX126x::startChannelScan(uint8_t symbolNum, uint8_t detPeak, uint8_t det
   RADIOLIB_ASSERT(state);
 
   // set mode to CAD
-  state = setCad(symbolNum, detPeak, detMin);
+  state = setCad(config.cad.symNum, config.cad.detPeak, config.cad.detMin, config.cad.exitMode, config.cad.timeout);
```

Comment:
Not sure if this comment ends up on the right line, but if the exit mode is CadToRx, then we would also need to configure the IRQs for Rx. For my usecase, I use PreambleDetected, may end up using HeaderOk, and of course Timeout and RxDone (not sure if all of these are covered by the combined Rx flags you have as defaults on some modules).

Reply:
Hmm, I'm thinking that the most clean way to do that would be to add an `irqFlags` field to `CADScanConfig_t`, and set the IRQ falgs based on that. What do you think?

Reply:
That works iff there is a translation from generalized flags to module-specific flags, just as there now is in the opposite direction for `checkIrq()`, otherwise it would not be portable to another module without changes.

Reply:
Side note that such a translation function would be an equal generalization of `irqRxDoneRxTimeout()` currently in use by the LW stack.

Reply:
The only other solution I can think of is to set the flags in the radio-specific implementation method, e.g. in `SX126x::startChannelScan` or `LR11x0::startChannelScan`, but that feels rather hacky.

> That works iff there is a translation from generalized flags to module-specific flags, just as there now is in the opposite direction for checkIrq()

Actually, that makes sense even as a standalone change, so I will do it like that.